### ISSUE TITLE
Question about data and mem usage

### ISSUE THREAD

**Issue by mxjmtxrm** [intel/auto-round#514]
Hi, I tried to quantize a long context model. I set seqlength larger than 8k even more, then the OOM error arised.
I notice the output of the block is saved in GPU, which probably takes up more than half of the memory. I think this is something that can be optimized, such as putting the output cache on the CPU. Does it make sense?
And does seqlength have a big impact on the quantization performance?

**Comment by wenhuach21** [comment]
Yes, that's one option. However, another challenge is that activations need to be retained for gradient computation during the backward pass. One way to alleviate this issue is to use activation checkpointing. For more details, please refer to torch.utils.checkpoint.

Alternatively, you can use the device_map option in AutoRound to distribute layers in block across different GPUs. For an example, please refer to the [DeepSeek-R1-int4-AutoRound-awq-asym model](https://huggingface.co/OPEA/DeepSeek-R1-int4-AutoRound-awq-asym#generate-the-model).

**Comment by mxjmtxrm** [comment]
I'll try it. Does seqlength have a big impact on the quantization performance? I've tried 2k and 4k and there's no noticeable difference. Is quantization insensitive to seq_length?

**Comment by mxjmtxrm** [comment]
I don't think device_map is helpful, as it won't reduce the memory usage of output cache. Take QWQ32b as an example, if I set seq_length 8k, num_sample=512, the memory usage is about 8192*5120*512*2=43G. It is too huge. I think it's still important to think about how to reduce the memory usage of output cache.

**Comment by wenhuach21** [comment]
> I don't think device_map is helpful, as it won't reduce the memory usage of output cache. Take QWQ32b as an example, if I set seq_length 8k, num_sample=512, the memory usage is about 8192x5120x512x2=43G. It is too huge. I think it's still important to think about how to reduce the memory usage of output cache.

I believe this is not the case. When low_gpu_mem_usage is enabled, the input and output caches should reside on the CPU, with only one batch of data loaded onto the GPU at a time. If this behavior is not functioning as expected, we will investigate and fix it as soon as possible.

**Comment by wenhuach21** [comment]
> I'll try it. Does seqlength have a big impact on the quantization performance? I've tried 2k and 4k and there's no noticeable difference. Is quantization insensitive to seq_length?

Sorry, we don't have such data, as the evaluation task is relatively short. We mainly used tasks like LongBench to test KV cache quantization. If you have any recommended tasks to evaluate, we're happy to test them when we have the bandwidth.
